heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -183,78 +183,43 @@ public PackingPlan pack() {
    * @return new packing plan
    */
   public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
-    int adjustments = 0;
     this.numContainers = currentPackingPlan.getContainers().size();
     resetState();
 
     int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
     increaseNumContainers(additionalContainers);
-    LOG.info(String.format("Allocated "
-            + "%s additional containers for repack bring the number of containers to %s.",
+    LOG.info(String.format(
+        "Allocated %s additional containers for repack bring the number of containers to %s.",
         additionalContainers, this.numContainers));
-    // Get the instances using Resource Compliant Round Robin allocation
-    Optional<Map<Integer, List<InstanceId>>> resourceCompliantRRAllocation =
-        getResourceCompliantRRAllocation(currentPackingPlan, componentChanges);
 
-    while (!resourceCompliantRRAllocation.isPresent()) {
-      //The number of containers has been updated
-      if (this.numAdjustments > adjustments) {
+    int adjustments = 0;
+    while (adjustments <= this.numAdjustments) {
+      try {
+        PackingPlanBuilder planBuilder = newPackingPlanBuilder(currentPackingPlan);
+        planBuilder.updateNumContainers(numContainers);
+        planBuilder = getResourceCompliantRRAllocation(
```

Comment:
If get resource compliant allocation fails with an exception, then the packingbuilder will still have the updaed number of containers? Shouldn't we bring it back to the old value in the catch clause? It might not make a difference eventually since we update again at the next iteration but it guarantees consistency. Now it is a little tricky to me to check whether the numContainers in the builder is always the correct one.